homeassistant/components/mobile_app/__init__.py

```diff
@@ -103,6 +103,28 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     registration_name = f"Mobile App: {registration[ATTR_DEVICE_NAME]}"
     webhook_register(hass, DOMAIN, registration_name, webhook_id, handle_webhook)
 
+    async def create_cloud_hook() -> None:
+        """Create a cloud hook."""
+        hook = await cloud.async_create_cloudhook(hass, webhook_id)
+        hass.config_entries.async_update_entry(
+            entry, data={**entry.data, CONF_CLOUDHOOK_URL: hook}
+        )
+
+    async def manage_cloudhook(state: cloud.CloudConnectionState) -> None:
+        if (
+            state is cloud.CloudConnectionState.CLOUD_CONNECTED
+            and CONF_CLOUDHOOK_URL not in entry.data
+        ):
+            await create_cloud_hook()
+
+    if (
+        CONF_CLOUDHOOK_URL not in registration
+        and cloud.async_active_subscription(hass)
+        and cloud.async_is_connected(hass)
+    ):
+        await create_cloud_hook()
+    entry.async_on_unload(cloud.async_listen_connection_change(hass, manage_cloudhook))
```

Comment:
In netatmo we only listen for cloud connection changes if there's an active cloud subscription.

Reply:
As discussed on discord, it's good to listen for connection changes regardless of the current subscription state.